src/tsx/coreConnection.tsx

```diff
@@ -1,31 +1,49 @@
+import { INodeCollection } from "./types";
+
 /**
  * This class handles the communication with the core.
  * The user calls methods directly to
  * send a message to the core, and can register for events.
  */
 interface ICoreConnection extends EventTarget {
   // UI-initiated actions
-  /** Example of UI-initiated action */
-  askForIncrement(): void;
+  handshake(): void;
+  nextNode(): void;
 
   // Events
-  addEventListener(event: "increment", listener: () => void): void;
+  addEventListener(event: "nodes",
+                   listener: (event: CustomEvent<INodeCollection>) => void): void;
+  addEventListener(event: "current-node",
+                   listener: (event: CustomEvent<string>) => void): void;
 }
 
 /**
  * This is a dummy implementation of a connection
  * to the core, for use in development.
  */
 class DummyCoreConnection extends EventTarget implements ICoreConnection {
+  private nodes: INodeCollection;
+  private currentNode: string;
 
   // We only use the address in the real core connection
-  // tslint:disable-next-line no-empty
   constructor(address: string) {
     super();
+    this.nodes = {
+      a: {next: "b", prompt: "Välkommen till Kulturkrocks senaste föreställning!"},
+      b: {next: "c", prompt: "Jag ser att ni fortfarande är på väg in."},
+      c: {next: "a", prompt: "Några av er kanske missade det, så jag säger det igen:"},
+    };
+    this.currentNode = "a";
+  }
+
+  public handshake() {
+    this.dispatchEvent(new CustomEvent("nodes", {detail: this.nodes}));
+    this.dispatchEvent(new CustomEvent("current-node", {detail: this.currentNode}));
   }
 
-  public askForIncrement() {
-    setTimeout(() => this.dispatchEvent(new Event("increment")), 1000);
+  public nextNode() {
+    this.currentNode = this.nodes[this.currentNode].next;
```

Comment:
Do we think it is easy to add multi-branching support on top of this?

Reply:
I think "next" and "choose path" will be separate actions, triggered by different keys, so this method will probably just need to check if we're at a branch point and do nothing/trigger an error if we are.

The structure of a node will probably need to change when we implement branching though. I think it's still simple enough that it will be easy to change the structure, but it might be worth it to come up with a format that works with branching already. What do you think?

Reply:
Sounds reasonable. I realized we actually could implement branching without even changing this. If we force the user to call selectBranch (or something like that) that sets the *next* field from a set of possible nodes first (we can initialize *next* to undefined I guess), nextNode will actually work out of the box with branching too.

Of course then we would have to be very careful with always calling selectBranch first and the node object wouldnt be immutable (as it seems otherwise to be) so there are of course drawbacks to that solution.

Anyhoot, I have been convinced this will work fine!